share/steel/examples/pulse/dice/dpe/DPE.fsti

```diff
@@ -0,0 +1,136 @@
+module DPE
+open FStar.Ghost
+open Steel.ST.Util
+open Steel.FractionalPermission
+open Pulse.Steel.Wrapper
+module W = Pulse.Steel.Wrapper
+open HACL
+open X509
+open EngineTypes
+open EngineCore
+open L0Types
+open L0Core
+module L = Steel.ST.SpinLock
+module A = Steel.ST.Array
+module R = Steel.ST.Reference
+module US = FStar.SizeT
+module U8 = FStar.UInt8
+module U32 = FStar.UInt32
+open Array
+open LinearScanHashTable
+open PulseHashTable
+module PHT = PulseHashTable
+module LSHT = LinearScanHashTable
+
+(* L1 Context -- no dedicated L1 logic, so there's no good place for this to live *)
+noeq
+type l1_context = { deviceID_priv: A.larray U8.t (US.v v32us);
+                    deviceID_pub: A.larray U8.t (US.v v32us);
+                    aliasKey_priv: A.larray U8.t (US.v v32us);
+                    aliasKey_pub: A.larray U8.t (US.v v32us);
+                    aliasKeyCRT: A.array U8.t;
+                    deviceIDCSR: A.array U8.t; }
+
+let l1_context_perm (c:l1_context)
```

Comment:
Do you need to expose this definition in the interface or can it be an abstract vprop for clients?

Reply:
A vprop's definition needs to be exposed in the interface (if there is one) in order for Pulse to fold/unfold the definition in modules that `open` the module containing the vprop. e.g. if the definition of [`PulseHashTable.ht_perm`](https://github.com/FStarLang/steel/blob/megan_pulse_lib/share/steel/examples/pulse/lib/PulseHashTable.fsti#L55) is moved from PulseHashTable.fsti to PulseHashTable.fst then we get the following error when checking DPE.fst:

[DPE.fst(49,25-52,33)](https://github.com/FStarLang/steel/blob/megan_pulse_lib/share/steel/examples/pulse/dice/dpe/DPE.fst#L49): (Error) cannot prove vprop PulseHashTable.ht_perm s ht in the context: PulseHashTable.models s ht pht **
emp

Reply:
this does seem like unexpected behavior, maybe it's worth opening an issue? let me know

Reply:
NOTE: this is **expected** behavior -- a module should not rely on the definition of a vprop in another module. some refactoring may be in order...